boot/freeldr/freeldr/arch/uefi/uefimem.c

```diff
@@ -0,0 +1,296 @@
+        case EfiMemoryMappedIOPortSpace:
+            return LoaderReserve;
+        default:
+            break;
+    }
+    return LoaderReserve;
+}
+
+PFREELDR_MEMORY_DESCRIPTOR
+UefiMemGetMemoryMap(ULONG *MemoryMapSize)
+{
+    EFI_LOADED_IMAGE_PROTOCOL *LoadedImage;
+    UINT32 DescriptorVersion;
+    SIZE_T FreeldrMemMapSize;
+    UINTN DescriptorSize;
+    EFI_STATUS Status;
+    UINTN MapSize;
+    UINTN MapKey;
+    UINT32 Index;
+
+    EFI_GUID EfiLoadedImageProtocol = EFI_LOADED_IMAGE_PROTOCOL_GUID;
+    PFREELDR_MEMORY_DESCRIPTOR FreeldrMem = NULL;
+    EFI_MEMORY_DESCRIPTOR* MapEntry = NULL;
+    UINT32 EntryCount = 0;
+    FreeldrDescCount = 0;
+
+    Status = GlobalSystemTable->BootServices->HandleProtocol(GlobalImageHandle,
+                                                             &EfiLoadedImageProtocol,
+                                                             (VOID**)&LoadedImage);
+    if (Status != EFI_SUCCESS)
+    {
+        TRACE("Failed to find LoadedImageHandlewith status: %d", Status);
+        UiMessageBoxCritical("Unable to initialize memory manager.");
+    }
+    OsLoaderBase = LoadedImage->ImageBase;
+    OsLoaderSize = LoadedImage->ImageSize;
+    PublicBootHandle = LoadedImage->DeviceHandle;
+    EfiMemoryMap = NULL;
+
+
+    TRACE("UefiMemGetMemoryMap: Gather memory map\n");
+    PUEFI_LoadMemoryMap(&MapKey,
+                        &MapSize,
+                        &DescriptorSize,
+                        &DescriptorVersion);
+
+    TRACE("Value of MapKey: %d\n", MapKey);
+    TRACE("Value of MapSize: %d\n", MapSize);
+    TRACE("Value of DescriptorSize: %d\n", DescriptorSize);
+    TRACE("Value of DescriptorVersion: %d\n", DescriptorVersion);
+
+    EntryCount = (MapSize / DescriptorSize);
+
+    FreeldrMemMapSize = (sizeof(FREELDR_MEMORY_DESCRIPTOR) * EntryCount);
+	Status = GlobalSystemTable->BootServices->AllocatePool(EfiLoaderData, FreeldrMemMapSize, (void**)&FreeldrMem);
+    if (Status != EFI_SUCCESS)
+    {
+        TRACE("Failed to allocate pool with status %d", Status);
+        UiMessageBoxCritical("Unable to initialize memory manager.");
+    }
+
+    RtlZeroMemory(FreeldrMem, FreeldrMemMapSize);
+    MapEntry = EfiMemoryMap;
+	for (Index = 0; Index < EntryCount; ++Index)
+    {
+        if (UefiConvertToFreeldrDesc(MapEntry->Type) == LoaderFree)
+        {
+            Status = GlobalSystemTable->BootServices->AllocatePages(AllocateAddress, EfiLoaderData,
+                                                                    MapEntry->NumberOfPages, &MapEntry->PhysicalStart);
+            if (Status == EFI_SUCCESS)
+            {
+                UefiSetMemory(FreeldrMem, MapEntry->PhysicalStart, MapEntry->NumberOfPages,
+                              LoaderFree);
+            }
+            else
+            {
+                UefiSetMemory(FreeldrMem, MapEntry->PhysicalStart, MapEntry->NumberOfPages,
+                              LoaderFirmwareTemporary);
```

Comment:
I would omit the line-break here. Especially since a few lines upwards (line241) you already created longer lines than that.